paasta_tools/cli/cmds/validate.py

```diff
@@ -209,6 +224,60 @@ def path_to_soa_dir_service(service_path):
     return soa_dir, service
 
 
+def validate_tron(service_path):
+    soa_dir, service = path_to_soa_dir_service(service_path)
+    returncode = True
+
+    if soa_dir.endswith('/tron'):
+        # Makes it possible to validate files in tron/ rather than service directories
+        # TODO: Clean up after migration to services is complete
+        cluster = service
+        soa_dir = soa_dir[:-5]
+        filenames = [filename for filename in os.listdir(service_path) if filename.endswith('.yaml')]
+        for filename in filenames:
+            namespace = os.path.splitext(filename)[0]
+            file_path = os.path.join(service_path, filename)
+            if not validate_schema(file_path, 'tron'):
+                returncode = False
+            if not validate_tron_namespace(namespace, cluster, soa_dir, tron_dir=True):
+                returncode = False
+    else:
+        # Normal service directory
+        for cluster in list_tron_clusters(service, soa_dir):
+            if not validate_tron_namespace(service, cluster, soa_dir):
+                returncode = False
+
+    return returncode
+
+
+def validate_tron_namespace(service, cluster, soa_dir, tron_dir=False):
+    returncode = True
+    job_configs, other_config = load_tron_service_config(service, cluster, soa_dir)
+
+    if tron_dir:
+        display_name = f'{cluster}/{service}.yaml'
+    else:
+        display_name = f'tron-{cluster}.yaml'
+
+    if service != MASTER_NAMESPACE and other_config:
+        returncode = False
+        other_keys = list(other_config.keys())
+        paasta_print(failure(
+            f"{display_name}: Non-{MASTER_NAMESPACE} namespace cannot have other config values, found {other_keys}",
+            "http://tron.readthedocs.io/en/latest/jobs.html",
+        ))
+
+    for job_config in job_configs:
+        check_msgs = job_config.validate()
+        if check_msgs:
+            returncode = False
+            paasta_print(invalid_tron_job(cluster, job_config.get_name(), "\n  ".join(check_msgs), display_name))
+        else:
+            if not tron_dir:
```

Comment:
What will happen if tron_dir?

Reply:
It only prints messages if something is invalid, not the successful validation messages. It was too noisy / easy to miss the invalid messages for tron prod otherwise.